src/Type/ImmutableArrayType.php

```diff
@@ -50,7 +50,8 @@ public function configureOptions(OptionsResolver $resolver): void
             'keys' => [],
         ]);
 
-        $resolver->setAllowedValues('keys', static function ($value): bool {
+        $resolver->setAllowedTypes('keys', 'iterable');
```

Comment:
Maybe would be nice to take a look at all types and try to narrow down the options, wdyt @VincentLanglet ?

Reply:
Indeed

Reply:
I think it should be done i another PR , I open an issue.